From 2addce77d72d6fc42902ddabfcfbedee729ee1ca Mon Sep 17 00:00:00 2001
From: Andre Heider <a.heider@gmail.com>
Date: Sun, 20 Nov 2022 16:52:42 +0100
Subject: [PATCH 19/39] nl80211: mark frequencies where HE operation in not
 allowed

Repurpose the IWINFO_FREQ_NO_2160MHZ define for that, which was
introduced but never used. nl80211 doesn't have such a flag either.

Signed-off-by: Andre Heider <a.heider@gmail.com>
---
 include/iwinfo.h | 2 +-
 iwinfo_nl80211.c | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/include/iwinfo.h b/include/iwinfo.h
index f2d6d35..2965785 100644
--- a/include/iwinfo.h
+++ b/include/iwinfo.h
@@ -66,7 +66,7 @@
 #define IWINFO_FREQ_NO_HT40MINUS	(1 << 3)
 #define IWINFO_FREQ_NO_80MHZ		(1 << 4)
 #define IWINFO_FREQ_NO_160MHZ		(1 << 5)
-#define IWINFO_FREQ_NO_2160MHZ		(1 << 6)
+#define IWINFO_FREQ_NO_HE		(1 << 6)
 
 extern const char *IWINFO_CIPHER_NAMES[IWINFO_CIPHER_COUNT];
 extern const char *IWINFO_KMGMT_NAMES[IWINFO_KMGMT_COUNT];
diff --git a/iwinfo_nl80211.c b/iwinfo_nl80211.c
index fba67d8..694a258 100644
--- a/iwinfo_nl80211.c
+++ b/iwinfo_nl80211.c
@@ -3029,6 +3029,8 @@ static int nl80211_get_freqlist_cb(struct nl_msg *msg, void *arg)
 						e->flags |= IWINFO_FREQ_NO_20MHZ;
 					if (freqs[NL80211_FREQUENCY_ATTR_NO_10MHZ])
 						e->flags |= IWINFO_FREQ_NO_10MHZ;
+					if (freqs[NL80211_FREQUENCY_ATTR_NO_HE])
+						e->flags |= IWINFO_FREQ_NO_HE;
 
 					e++;
 					arr->count++;
-- 
2.35.1

